src/UrlChecker/DefaultUrlChecker.php

```diff
@@ -102,6 +102,8 @@ protected function _getChecker(array $options = [])
      */
     protected function _getUrlFromRequest(UriInterface $uri, $getFullUrl = false)
     {
+        $uri = $uri->withPath($uri->base . $uri->getPath());
```

Comment:
Would an application in a base directory have it `loginUrl` setup to have the base path?

Reply:
I'm not clear on the question. Is your concern that `$uri->base` will never be non-empty here, or that the resulting URL will have the base duplicated?

Reply:
My question is more on how loginUrl is generally used for applications in subdirectories. Is it configured to be `/base/login` or `/login`?

Reply:
Just realized that your concern may be that if we make this URL include the base, but the configured one didn't have that as part of it, then they won't match. I guess my response would be that, since this already isn't working in subfolders, there shouldn't really be any code out there relying on any particular behaviour. And since the URL checker does strict comparisons without any type conversion, it has to be given string URLs in the configuration, and the sensible way of generating those strings is with `Router::url()`, which will include the base.

In short, including the base here shouldn't break existing code, and works as expected with `Router::url()`.